statsmodels/base/model.py

```diff
@@ -510,6 +510,164 @@ def hess(params, *args):
         mlefit.mle_settings = optim_settings
         return mlefit
 
+    def _fit_zeros(self, keep_index=None, start_params=None,
+                   return_auxiliary=False, k_params=None, **fit_kwds):
+        """experimental fit the model subject to zero constraints
+
+        Intended for internal use cases until we know what we need.
+        API will need to change to handle models with two exog.
+        This is not yet supported by all model subclasses.
+
+        This is essentially a simplified version of `fit_constrained`, and
+        does not need to use `offset`.
+
+        The estimation creates a new model with transformed design matrix,
+        exog, and converts the results back to the original parameterization.
+
+        Some subclasses could use a more efficient calculation than using a
+        new model.
+
+        Parameters
+        ----------
+        keep_index : array_like (int or bool) or slice
+            variables that should be dropped.
+        start_params : None or array_like
+            starting values for the optimization. `start_params` needs to be
+            given in the original parameter space and are internally
+            transformed.
+        k_params : int or None
+            If None, then we try to infer from start_params or model.
+        **fit_kwds : keyword arguments
+            fit_kwds are used in the optimization of the transformed model.
+
+        Returns
+        -------
+        results : Results instance
+
+        """
+        # we need to append index of extra params to keep_index as in
+        # NegativeBinomial
+        if hasattr(self, 'k_extra') and self.k_extra > 0:
+            # we cannot change the original, TODO: should we add keep_index_params?
+            keep_index = np.array(keep_index, copy=True)
+            k = self.exog.shape[1]
+            extra_index = np.arange(k, k + self.k_extra)
+            keep_index_p = np.concatenate((keep_index, extra_index))
+        else:
+            keep_index_p = keep_index
+
+        # not all models support start_params, drop if None, hide them in fit_kwds
+        if start_params is not None:
+            fit_kwds['start_params'] = start_params[keep_index_p]
+            k_params = len(start_params)
+            # ignore k_params in this case, or verify consisteny?
```

Comment:
"TODO: "?  the question mark makes it pretty clear in context, but its really nice to be able to grep for these

Reply:
I leave it as is, I don't remember which models this affects.
i.e. this means that this part might be fragile and might need investigation if it fails for specific cases.